torch/csrc/dynamo/eval_frame.c

```diff
@@ -320,6 +320,39 @@ inline static CacheEntry* get_cache_entry(PyCodeObject* code) {
   return extra;
 }
 
+PyObject* _debug_get_cache_entry_list(PyObject* self, PyObject* args) {
+  PyObject* object;
+  if (!PyArg_ParseTuple(args, "O", &object)) {
+    return NULL;
+  }
+  if (!PyCode_Check(object)) {
+    PyErr_SetString(PyExc_TypeError, "expected a code object!");
+  }
+  PyCodeObject* code = (PyCodeObject*)object;
+
+  CacheEntry* current_node = get_cache_entry(code);
+
+  PyObject* outer_list = PyList_New(0);
+  if (!outer_list) {
+    return NULL;  // Return NULL if failed to create list
+  }
+  while (current_node != NULL) {
+    // Creating a new Python tuple for the check_fn and code of current CacheEntry
+    PyObject* inner_list = PyTuple_Pack(2, current_node->check_fn, current_node->code);
+    // Add the inner list to the outer list
+    if (PyList_Append(outer_list, inner_list) < 0) {
+      Py_DECREF(outer_list);
+      Py_DECREF(inner_list);  // Clean up if failed to append
+      return NULL;
+    }
+
```

Comment:
I think you need a `Py_DECREF(inner_list)` here, since you still own a reference to `inner_list`.  `PyList_Append` will add a reference, so I'd expect the refcount to be 2 here.

Reply:
Fixed in [514a391](https://github.com/pytorch/pytorch/pull/106673/commits/514a3916d2409eae928b3a17ec0d786e6a26c448). `Py_DECREF(inner_list)` after calling `PyList_Append` anyway.